src/app/modules/submission/email-submission/__tests__/email-submission.routes.spec.ts

```diff
@@ -0,0 +1,368 @@
+}
+
+const EmailSubmissionsRouter = Router()
+EmailSubmissionsRouter.post(
+  SUBMISSIONS_ENDPT,
+  CaptchaFactory.validateCaptchaParams,
+  FormController.formById,
+  PublicFormMiddleware.isFormPublicCheck,
+  CaptchaMiddleware.checkCaptchaResponse as RequestHandler,
+  SpcpController.isSpcpAuthenticated,
+  EmailSubmissionsMiddleware.receiveEmailSubmission,
+  celebrate({
+    body: Joi.object({
+      responses: Joi.array()
+        .items(
+          Joi.object()
+            .keys({
+              _id: Joi.string().required(),
+              question: Joi.string().required(),
+              fieldType: Joi.string()
+                .required()
+                .valid(...Object.values(BasicField)),
+              answer: Joi.string().allow(''),
+              answerArray: Joi.array(),
+              filename: Joi.string(),
+              content: Joi.binary(),
+              isHeader: Joi.boolean(),
+              myInfo: Joi.object(),
+              signature: Joi.string().allow(''),
+            })
+            .xor('answer', 'answerArray') // only answer or answerArray can be present at once
+            .with('filename', 'content'), // if filename is present, content must be present
+        )
+        .required(),
+      isPreview: Joi.boolean().required(),
+    }),
+  }),
+  EmailSubmissionsMiddleware.validateEmailSubmission,
+  MyInfoController.verifyMyInfoVals as RequestHandler,
+  (SubmissionsController.injectAutoReplyInfo as unknown) as RequestHandler,
+  SpcpController.appendVerifiedSPCPResponses as RequestHandler,
+  EmailSubmissionsMiddleware.prepareEmailSubmission as RequestHandler,
+  EmailSubmissionsMiddleware.saveMetadataToDb,
+  EmailSubmissionsMiddleware.sendAdminEmail,
+  SubmissionsController.sendAutoReply,
+)
+
+const EmailSubmissionsApp = setupApp('/', EmailSubmissionsRouter)
+
+describe('email-submission.routes', () => {
+  let request: Session
+
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(() => {
+    request = session(EmailSubmissionsApp)
+  })
+  afterEach(async () => {
+    await dbHandler.clearDatabase()
+    jest.restoreAllMocks()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('SPCP authentication', () => {
+    describe('SingPass', () => {
+      it('should return 200 when submission is valid', async () => {
+        mockSpClient.verifyJWT.mockImplementationOnce((_jwt, cb) =>
+          cb(null, {
+            userName: 'S1234567A',
+          }),
+        )
+        const { form } = await dbHandler.insertEmailForm()
+        // authType cannot be saved if form is published, so need to save twice
+        form.esrvcId = 'mockEsrvcId'
+        form.authType = AuthType.SP
+        form.hasCaptcha = false
+        await form.save()
+        form.status = Status.Public
+        await form.save()
+
+        const response = await request
+          .post(`${SUBMISSIONS_ENDPT_BASE}/${form._id}`)
+          .field('body', JSON.stringify(MOCK_SUBMISSION_BODY))
```

Comment:
should `.send` be used? It will automatically be set on the body

```ts
.send(MOCK_SUBMISSION_BODY)
```

Reply:
I tried it but it didn't work, probably something to do with the content type being multipart form data. I'll try again just in case

Reply:
nope didn't work! this is why I used `.field` for email mode but `.send` for storage mode

Reply:
gotcha, np!